infor-design/enterprise
# 269 - Toolbar Flex: Better searchfield integration

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR addresses some of the points on #269:

- Adds a responsive view for an active searchfield residing inside a Flex Toolbar.  When active, the searchfield will expand to fill the space provided by the toolbar element and temporarily hide all the other items.  This replicates a feature present in the old toolbar.
- Completes the expand/collapse feature that was present on old Toolbar Searchfields on a Flex Toolbar.  When an "expanded" or "non-collapsible" searchfield is displayed below the tablet breakpoint (about 767px), it will collapse and appear as a search button until activated, which will cause it to expand to the newly added responsive view.

**Related github/jira issue (required)**:
#269 

**Steps necessary to review your pull request (required)**:
Pull this branch, build, run the demo app, then...

_To test responsive unopened state:_
- Navigate to http://localhost:4000/components/toolbar-flex/example-index?nofrills=true (don't test the regular view, since it's header still uses an old toolbar)
- Ensure all Searchfields on the page appear fully expanded with the correct background color
- Shrink the width of your browser viewport to less than 767px
- Ensure all Searchfields appear as a Search icon with a transparent background color.

_To test opened (focused) state:_
- With the browser viewport shrunk below 767px, focus one of the Search "icons"
- See that the other toolbar items disappear and the searchfield fills out to 100% of the width of the toolbar.
- Key in a search term.  Ensure that the clear button is in the correct spot.
- Defocus the searchfield.  Ensure that on the responsive view, no text is present on or near the search icon.
- Stretch your browser viewport to beyond 767px.
- Ensure that the searchfield has visible text entry, and it's gray background returns.
